flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -662,13 +662,13 @@ private void addComputationState(
 			ComputationState computationState) {
 		final Stack<State<T>> statesToCheck = new Stack<>();
 		statesToCheck.push(state);
-
+		ConditionContext<T> context = new ConditionContext<T>(this, sharedBuffer, computationState);
```

Comment:
How about creating the context only once at the very beginning of computeNextStates?